### PR TITLE
Clean up charts logic

### PR DESCRIPTION
**Pull request by tleraj**
The purpose of this PR is to close https://github.com/aqualinkorg/aqualink-app/issues/678.
With this approach, we introduce a new `Dataset` interface, with the following properties:

```typescript
  label: string;
  data: SofarValue[];
  type: "line" | "scatter";
  unit: string;
  curveColor: string;
  threshold?: number;
  fillColor?: string;
  fillColorAboveThreshold?: string;
  fillColorBelowThreshold?: string;
  surveysAttached?: boolean;
  considerForXAxisLimits?: boolean;
  maxHoursGap?: number;
  tooltipMaxHoursGap?: number;
  isDailyUpdated?: boolean;
  displayData?: boolean;
  cardColumnName?: string;
  cardColumnTooltip?: string;
  tooltipLabel?: string;
```

and we simply pass a `Dataset[]` prop to each chart we want to display.

### TODOs
 - [x] QA
 - [x] Hunt down and remove unused utils
 - [x] ~~Fix available ranges not showing bug (probably introduced in https://github.com/aqualinkorg/aqualink-app/pull/675)~~ Fixed by https://github.com/aqualinkorg/aqualink-app/pull/675/commits/2b5cd26cf9ef2378e1402f10a4fc0c0db888ffbb

### CODE REVIEW

File: packages/website/src/common/Chart/Tooltip/index.test.tsx [hunk 1]
```diff
@@ -8,12 +8,13 @@ test("renders as expected", () => {
...
-      spotterTopTemp={10}
-      spotterBottomTemp={15}
-      hoboBottomTemp={20}
+      // depth={10}
+      // historicalMonthlyMeanTemp={20}
+      // satelliteTemp={20}
+      // spotterTopTemp={10}
+      // spotterBottomTemp={15}
+      // hoboBottomTemp={20}
+      datasets={[]}
```
Comment (ericboucher) [inline packages/website/src/common/Chart/Tooltip/index.test.tsx:17]: is this expected?

File: packages/website/src/common/Chart/MultipleSensorsCharts/helpers.ts [hunk 2]
```diff
@@ -226,3 +243,157 @@ export const constructOceanSenseDatasets = (
     id: "oxidation_reduction_potential",
   },
 });
+
+export const hasAtLeastNData = (n: number, data?: SofarValue[]) =>
+  typeof data?.length === "number" && data.length >= n;
+
+export const generateTempAnalysisDatasets = (
```
Comment (ericboucher) [inline packages/website/src/common/Chart/MultipleSensorsCharts/helpers.ts:250]: I think this function can be simplified more

**Review by ericboucher — COMMENTED** [review]
Great work @tleraj 

While we are doing these big changes, let's make sure that you add docstring and comments throughout the code to make it easier to navigate in the future.

File: packages/website/src/common/Chart/MultipleSensorsCharts/helpers.ts [hunk 3]
```diff
@@ -141,17 +153,22 @@ export const findDataLimits = (
...
-    case hasSpotterData:
-      return "spotter";
-    case hasHoboData:
-      return "hobo";
+    case nCardColumns === 3:
+      return "small";
+    case nCardColumns === 2:
+      return "large";
+    case nCardColumns === 1:
+      return "large";
```
[the remark this answers is not in the document]
Comment (tleraj) [reply packages/website/src/common/Chart/MultipleSensorsCharts/helpers.ts:169]: For the case `nCardColumns === 2`, we ahould return `"medium"`. Sorry my bad.

File: packages/website/src/common/Chart/MultipleSensorsCharts/helpers.ts (hunk 2 above)
Comment (tleraj) [reply packages/website/src/common/Chart/MultipleSensorsCharts/helpers.ts:248]: I did some work in https://github.com/aqualinkorg/aqualink-app/pull/680/commits/c59cfb02a72c0c208a38506aa265555b7415021d. Let me know what you think :slightly_smiling_face:

File: packages/website/src/common/Chart/Tooltip/index.test.tsx (hunk 1 above)
Comment (tleraj) [reply packages/website/src/common/Chart/Tooltip/index.test.tsx]: No, I added a `mockDataset` and removed these comments.

File: packages/website/src/common/Chart/MultipleSensorsCharts/helpers.ts [hunk 4]
```diff
@@ -396,6 +422,14 @@ export const generateMetricDataset = (
   };
 };
 
+/**
+ * A util function to generate the daily data dataset, used in a bunch of places.
```
Comment (ericboucher) [inline packages/website/src/common/Chart/MultipleSensorsCharts/helpers.ts:426]: @tleraj let's keep the tone more formal. `used in a bunch of places` -> `used in "example" and other places`

File: packages/website/src/common/Chart/utils.ts [hunk 5]
```diff
@@ -145,6 +145,20 @@ export const convertHistoricalMonthlyMeanToSofar = (
 ): SofarValue[] | undefined =>
   data?.map((item) => ({ timestamp: item.date, value: item.value }));
 
+// // Util function used to fill the area between the chart's plot and the y axis 0.
```
Comment (ericboucher) [inline packages/website/src/common/Chart/utils.ts:148]: //

**Review by ericboucher — APPROVED** [review]

**Pull request merged**